src/main/java/ui/RoomView.java

```diff
@@ -1,26 +1,77 @@
 package ui;
 
 import leave_room.LeaveRoomController;
-
+import messaging.MessageController;
 import javax.swing.*;
 import java.awt.*;
 
+
 public class RoomView extends View {
     private final LeaveRoomController leaveRoomController;
+    private final MessageController sendMessageController;
+    private JTextField messageTextField;
+    private JTextArea chatTextArea;
 
-    public RoomView(LeaveRoomController leaveRoomController){
+    public RoomView(LeaveRoomController leaveRoomController, MessageController sendMessageController){
         this.leaveRoomController = leaveRoomController;
+        this.sendMessageController = sendMessageController;
     }
 
     @Override
     protected JPanel createPanel() {
-        JPanel panel = new JPanel(new GridLayout(3, 2, 10, 10));
+        JPanel panel = new JPanel(new BorderLayout());
         panel.setBorder(BorderFactory.createEmptyBorder(10, 10, 10, 10));
 
+        // Set up the JFrame
+        chatTextArea = new JTextArea(20, 50);
+        chatTextArea.setText(messageHistory);
+        chatTextArea.setLineWrap(true); // Enable line wrapping
+        chatTextArea.setEditable(false);
+
+        // Scroll through texts
+        JScrollPane scrollPane = new JScrollPane(chatTextArea);
+        scrollPane.setVerticalScrollBarPolicy(JScrollPane.VERTICAL_SCROLLBAR_ALWAYS);
+
+        // Create a JTextField to enter messages
+        messageTextField = new JTextField(30);
+
+
+
+        // Create a SendMessage
+        JButton sendMessageButton = new JButton("Send Message");
+        sendMessageButton.addActionListener(e -> sendMessageController.handleSendMessage(roomID, userID, messageTextField.getText()));
+
+
+        // Send Lark Button
+        JButton sendLarkButton = new JButton("Send Lark");
```

Comment:
Should comment out button until feature is implemented

Reply:
ok